enhancements/kubelet/user-namespaces-support.md

```diff
@@ -0,0 +1,339 @@
+
+### API Extensions
+
+- Add `UserNamespacesPolicy` to SCC. This relies on approval from the apiserver team.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+N/A
+
+#### Standalone Clusters
+
+N/A
+
+#### Single-node Deployments or MicroShift
+
+N/A
+
+### Implementation Details/Notes/Constraints
+
+#### SecurityContextConstraint Updates
+
+##### ``UserNamespacesPolicy`` field
+
+Add the following to the [Openshift API](https://github.com/openshift/api/blob/76a71dac36a08eab1b240c6c8d4e39c813b1b12b/security/v1/types.go):
+
+```
+       // userNamespacesPolicy determines if the policy allows host users in containers.
+       // Valid values are "AllowHostNamespaces", "RequirePodNamespaces", and omitted.
+       // When omitted, this means no opinion and the platform is left to choose a reasonable default,
+       // which is subject to change over time.
+       // The current default is "AllowHostNamespaces".
+       // +openshift:enable:FeatureGate=UserNamespacesPodSecurityStandards
+       // +kubebuilder:validation:Enum="AllowHostNamespaces";"RequirePodNamespaces";""
+       // +optional
+       `UserNamespacesPolicy` NamespacesStrategyType `json:"userNamespacesPolicy" protobuf:"bytes,26,opt,name=userNamespacesPolicy"`
+...
+// NamespacesStrategyType shows the allowable values for the UserNamespacesPolicy field.
+type NamespacesStrategyType string
+...
+const (
+       // NamespacesStrategyTypeAllowHost allows a pod to set `hostUsers` field to either `true` or `false`
+       NamespacesStrategyTypeAllowHost NamespacesStrategyType = "AllowHostNamespaces"
+       // NamespaceStrategyRequirePod requires the `hostUsers` field be `false` in a pod.
+       NamespaceStrategyRequirePod NamespacesStrategyType = "RequirePodNamespaces"
+       // NamespaceStrategyEmpty will set to the default, which is currently "AllowHostNamespaces".
+       NamespaceStrategyEmpty NamespacesStrategyType = ""
+
+...
+)
+
+```
+
+This field branches from its fellow `AllowHost*` namespace fields to follow current API conventions, which maintain boolean fields are not extendable enough.
+
+##### restricted-v3
+
+This SCC profile will be identical to the existing restricted-v2, except it will set the `UserNamespacesPolicy` to `RequirePodNamespaces`,
+thus forcing pods to be in a user namespace. This will make it a more restrictive profile, as the user on the host will not be
+the same as the one inside the container.
+
+Note: upstream Kubernetes deemed `runAsUser` to be a safe field to use for pods with `hostUsers: false`. However, the philosophy
+of SCC utilizes defense-in-depth. `restricted-v3` will be the most restrictive policy.
+
+##### nonroot-v3
+
+This SCC profile will be identical to `nonroot-v2`, except it will set `UserNamespacesPolicy` to `RequirePodNamespaces`, thus forcing pods to be in a
+user namespace. It differs from `restricted-v3` because it allows the fields `runAsUser`/`runAsGroup`/`fsGroup`.
```

Comment:
Wouldn't it be `runAsAny`?

`nonroot-v3` would have be like `restricted-v2` with user values set to `runAsAny` as in `anyuid`.

Reply:
updated and clarified